airflow/www/api/experimental/endpoints.py

```diff
@@ -269,6 +270,40 @@ def dag_run_status(dag_id, execution_date):
     return jsonify(info)
 
 
+@csrf.exempt
+@api_experimental.route('/dags/<string:dag_id>/cancel_dag_run', methods=['POST'])
+@requires_authentication
+def cancel_dag_run(dag_id):
+    """
+    Cancel a specific dagrun run_id for a given DAG
+    """
+    data = request.get_json(force=True)
+
+    run_id = None
+    if 'run_id' in data:
+        run_id = data['run_id']
+    else:
+        error_message = 'Missing run_id'
+        _log.info(error_message)
+        response = jsonify({'error': error_message})
+        response.status_code = 400
+        return response
+
+    try:
+       dr = _cancel_dag_run(dag_id, run_id)
+    except AirflowException as err:
+        _log.error(err)
+        response = jsonify(error="{}".format(err))
+        response.status_code = err.status_code
+        return response
+
+    if getattr(g, 'user', None):
```

Comment:
Will this ever not have a user? The authentication should make there be one no?

Reply:
it does if anonymous access is allowed. We don't use it but since this is in airflow codebase, we should support possibility as it will make it easier to merge this upstream